### PR TITLE
etcd: make commit queue test deterministic

### PR DESCRIPTION
**Pull request by bhandras**
Sometimes commit queue test fails due to timing issue introduced by sleeps. This PR changes the test to make it deterministic.

### CODE REVIEW

File: channeldb/kvdb/etcd/commit_queue_test.go [hunk 1]
```diff
@@ -68,48 +69,51 @@ func TestCommitQueue(t *testing.T) {
...
 	q.Add(
-		commit("blocked2", false),
-		makeReadSet([]string{"key2"}),
+		commit("blocked2", nil, nil),
+		makeReadSet([]string{"key2", "key5"}),
 		makeWriteSet([]string{"key4"}),
 	)
 
+	// Allow Tx1 to continue with the commit.
+	block.Unlock()
```
Comment (cfromknecht) [inline channeldb/kvdb/etcd/commit_queue_test.go:107]: nit: might be simpler to use a `ready` channel, e.g.:
```golang
commit := func(...) {
        ...
        if ready != nil {
              <-ready
        }
}
...
ready := make(chan struct{})
...
commit(_, _, ready)
...
close(ready)
```

This approach will also work with more than one "free" transaction

File: channeldb/kvdb/etcd/commit_queue_test.go [hunk 2]
```diff
@@ -68,48 +69,51 @@ func TestCommitQueue(t *testing.T) {
...
-		makeWriteSet([]string{"key3"}),
-	)
-	// Tx3: reads: key1, writes: key4, conflict: none
+	<-commitTx1
+
+	// Tx2: reads: key1, key5, writes: key3, conflict: Tx1 (key3)
+	// We don't expect queue add to block as this txn will queue up after
+	// tx1.
 	q.Add(
-		commit("free", true),
```
Comment (cfromknecht) [inline channeldb/kvdb/etcd/commit_queue_test.go:87]: This test seems a bit weaker now, in that it doesn't actually test for transaction reordering. If the commit queue simply executed the transactions in the order they are added, the test would pass. The new version doesn't test that concurrent non-conflicting transactions are executed immediately.

I think it makes sense to keep the current transactions, possibly even making the test stronger by naming the transactions `free1` and `free2` to assert their order.

File: channeldb/kvdb/etcd/commit_queue_test.go (hunk 1 above)
Comment (bhandras) [reply channeldb/kvdb/etcd/commit_queue_test.go:107]: See my other comment. We can use a channel, but in the test example it's not possible to submit overlapping conflicting/non-conflicting txns without introducing a deadlock in the test.

File: channeldb/kvdb/etcd/commit_queue_test.go [hunk 3]
```diff
@@ -7,24 +7,25 @@ import (
...
 	var wg sync.WaitGroup
 	commits := make([]string, numCommits)
 	idx := int32(-1)
 
-	commit := func(tag string, sleep bool) func() {
+	commit := func(tag string, commit chan struct{},
+		block *sync.Mutex) func() {
+
 		return func() {
+			if commit != nil {
```
Comment (halseth) [inline channeldb/kvdb/etcd/commit_queue_test.go:26]: can names this `signalStart` or something other that is more descriptive than `commit`

File: channeldb/kvdb/etcd/commit_queue_test.go [hunk 4]
```diff
@@ -33,8 +34,8 @@ func TestCommitQueue(t *testing.T) {
 			i := atomic.AddInt32(&idx, 1)
 			commits[i] = tag
 
-			if sleep {
-				time.Sleep(commitDuration)
+			if block != nil {
```
Comment (halseth) [inline channeldb/kvdb/etcd/commit_queue_test.go:37]: this needs a comment, same for the channel close above

File: channeldb/kvdb/etcd/commit_queue_test.go (hunk 1 above)
Comment (halseth) [reply channeldb/kvdb/etcd/commit_queue_test.go:107]: I agree that using a channel to signal that the commit can continue is better than a mutex

**Comment by bhandras** [comment]
Closing this as another solution has been implemented in: https://github.com/lightningnetwork/lnd/pull/5513

**Pull request closed without merging**